src/main/java/org/testng/internal/ConfigInvoker.java

```diff
@@ -263,51 +259,62 @@ public void invokeConfigurations(
           log(3, "Skipping " + Utils.detailedMethodName(tm, true) + " because it is not enabled");
           continue;
         }
-        if (hasConfigurationFailureFor(currentTestMethod, tm.getGroups() , testClass, instance) && !alwaysRun) {
+        if (hasConfigurationFailureFor(configMethodAttributes.getTestMethod(), tm.getGroups() ,
```

Comment:
Next step will be to directly pass configMethodAttributes to methods? :)

Reply:
Hopefully yes. But I will do that in the next PR as well.